### PR TITLE
Bug fixed when reading data sets from library

### LINKED ISSUE #6969: Reading more data sets from the R library
My latest example is the climatol package I have added recently to evaluate.  It has a number of datasets.  Here is a snapshot.
image

You will notice that there is a name for the file and then another name in the brackets.  Currently the code to read the first file is:

```
# Code generated by the dialog, Open Dataset from Library
utils::data(package="climatol", X=dah)

data_book$import_data(data_tables=list(dah=dah))
```
This gives an error and fails to read.  I can change it to read the name in the brackets, as follows:
```
# Code run from Script Window
utils::data(package="climatol", X=pcp_results)

data_book$import_data(data_tables=list(pcp_results=dah))
```
This imports fine.
I suggest changing the reading to the name in the brackets, as above, when it finds a bracket.  (I am slightly confused that it uses both parts of the name - but then my R is really flaky!

### PR DESCRIPTION
**Pull request by N-thony**
Fixes #6969
@rdstern this fixes the bug that was appearing when trying to load the data library especially in the case described in the corresponding issue. Could you take a look? Thanks.
@africanmathsinitiative/developers this is ready for review. Thanks.

### CODE REVIEW

File: instat/dlgFromLibrary.vb [hunk 1]
```diff
@@ -218,6 +218,16 @@ Public Class dlgFromLibrary
         End If
     End Function
 
+    Private Function GetStringInBracket(ByRef strValue As String)
+        Dim startIndex As Integer = strValue.IndexOf("(") + 1
+        Dim strLength As Integer = strValue.IndexOf(")") - startIndex
```
Comment (lloyddewit) [inline instat/dlgFromLibrary.vb:223]: R-Instat naming convention for integers is `iStartIndex` etc.

**Comment by N-thony** [comment]
> Fixes #6969 @rdstern this fixes the bug that was appearing when trying to load the data library especially in the case described in the corresponding issue. Could you take a look? Thanks.

**Review by lloyddewit — APPROVED** [review]

File: instat/dlgFromLibrary.vb (hunk 1 above)
Comment (N-thony) [reply instat/dlgFromLibrary.vb]: @lloyddewit I have made the change.

File: instat/dlgFromLibrary.vb [hunk 2]
```diff
@@ -218,6 +218,16 @@ Public Class dlgFromLibrary
         End If
     End Function
 
+    Private Function GetStringInBracket(ByRef strValue As String)
+        Dim iStartIndex As Integer = strValue.IndexOf("(") + 1
+        Dim strLength As Integer = strValue.IndexOf(")") - iStartIndex
```
Comment (lloyddewit) [inline instat/dlgFromLibrary.vb:223]: naming convention?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Review by lloyddewit — APPROVED** [review]

**Review by rdstern — APPROVED** [review]
Nice.  I also tested on the survey package that has a similar situation and they now read ok too.

**Pull request merged**